File: Intermediate/17.Find the first non-repeating character in a string/Solution.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;

char firstNonRepeatingChar(const string& str) {
    unordered_map<char, int> count;

    for (char c : str) {
        count[c]++;
    }

    for (char c : str) {
        if (count[c] == 1) {
            return c;
        }
    }
    return '\0';
}

int main() {
    string str;
    cout << "Enter a string: ";
    cin >> str;

    char result = firstNonRepeatingChar(str);
    if (result != '\0') {
        cout << "The first non-repeating character is: " << result << endl;
    } else {
        cout << "No non-repeating character found.\n";
    }

    return 0;
}

File: Intermediate/17.Find the first non-repeating character in a string/Solution.py
def first_non_repeating_char(s):
    from collections import Counter
    count = Counter(s)

    for char in s:
        if count[char] == 1:
            return char
    return None

if __name__ == "__main__":
    s = input("Enter a string: ")
    result = first_non_repeating_char(s)
    if result:
        print(f"The first non-repeating character is: {result}")
    else:
        print("No non-repeating character found.")
